### PR TITLE
Components

### PR DESCRIPTION
**Pull request by Katsiaryna-Andrabaila**
1. Task: https://github.com/rolling-scopes-school/tasks/tree/master/react/modules/module01
2. Screenshot: 
components
3. Deploy: https://katsiaryna-andrabaila-react.netlify.app/
4. Done 19.03.2023 / deadlne 20.03.2023
5. Score 15 / 15
6.  Commands:
     + To go to `components` branch use "git checkout components"

     + To check test coverage use "npm run test"

     + To check lint errors use "npm run lint"

     + To run remote server use "npm start"

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  "dependencies": {
+    "@babel/plugin-syntax-flow": "^7.18.6",
+    "@babel/preset-flow": "^7.18.6",
+    "@testing-library/jest-dom": "^5.16.5",
+    "@testing-library/react": "^13.4.0",
+    "@testing-library/user-event": "^14.0.0-beta",
+    "@types/jest": "^27.5.2",
+    "@types/node": "^16.18.14",
+    "@types/react": "^18.0.28",
+    "@types/react-dom": "^18.0.11",
```
Comment (alehharnizonau) [inline package.json]: These packages should be installed as devDependencies. Please be familiar with https://www.geeksforgeeks.org/difference-between-dependencies-devdependencies-and-peerdependencies/ and reorder packages

File: src/components/footer/Footer.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  render() {
+    return (
+      <footer className="footer">
+        <a
+          href="https://github.com/Katsiaryna-Andrabaila"
+          target="_blank"
+          rel="noreferrer"
+          className="github-link"
+        >
+          katsiaryna-andrabaila
```
Comment (alehharnizonau) [inline src/components/footer/Footer.tsx]: It looks like magic string)

File: src/components/searchBar/SearchBar.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  render() {
+    return (
+      <div className="search-wrapper">
+        <input
+          type="search"
+          className="search-bar"
+          value={this.state.value}
+          onChange={(event) => {
+            this.setState({ value: event.target.value });
+            this.saveValue(event.target.value);
```
Comment (alehharnizonau) [inline src/components/searchBar/SearchBar.tsx]: Better to use functions or methods in our case for that, e.g. handleChange.

**Review by alehharnizonau — COMMENTED** [review]
Good job! There are some moments, I commented. Also for tests you can use tests naming convention. See more there https://infinum.com/handbook/frontend/react/testing-best-practices#:~:text=Naming%20Convention&text=Root%20describe%20must%20be%20the,be%20a%20use%20case%20sentence

File: package.json (hunk 1 above)
Comment (Katsiaryna-Andrabaila) [reply package.json]: 9aa2b92a436a154213de45a39df79ee1c7b152f1

File: src/components/footer/Footer.tsx (hunk 2 above)
Comment (Katsiaryna-Andrabaila) [reply src/components/footer/Footer.tsx]: 11e1c25c1fd6fb834633138d85e065d4eead2c14

File: src/components/searchBar/SearchBar.tsx (hunk 3 above)
Comment (Katsiaryna-Andrabaila) [reply src/components/searchBar/SearchBar.tsx]: f126eb21865e77308cbb7cf888c2e98d9d7c9f77

**Review by alehharnizonau — APPROVED** [review]

**Pull request merged**